sklearn/discriminant_analysis.py

```diff
@@ -529,14 +529,13 @@ def predict_proba(self, X):
         C : array, shape (n_samples, n_classes)
             Estimated probabilities.
         """
-        prob = self.decision_function(X)
-        expit(prob, out=prob)
-        if len(self.classes_) == 2:  # binary case
-            return np.column_stack([1 - prob, prob])
+        check_is_fitted(self, 'classes_')
+
+        if len(self.classes_) == 2:
```

Comment:
Thinking a bit more, I would avoid calling the base class implementation of the logisitic regression just for the 2d case:

```python
decision = self.decision_function(X)
if self.classes_.size == 2:
    proba = expit(decision, out=decision)
    return np.vstack([1 - proba, proba]).T
else:
    return softmax(decision)
```

Reply:
This is not calling a `LogisticRegression` method, but rather the `LinearClassifierMixin` method `_predict_proba_lr` (which, of course, `LogisticRegression` also inherits). 

The code you are suggesting is essentially duplicating this very method.
https://github.com/scikit-learn/scikit-learn/blob/af842d3680f46adec8841d5444591c8bc4adf701/sklearn/linear_model/base.py#L287-L301

Are you sure this is what we want?

BTW, I am happy to change `len(self.classes_)` to `self.classes_.size` if this style is preferred.

Reply:
> Are you sure this is what we want?

Actually, I want exactly this. The reason is that this `_predict_proba_lr` corresponds to the OvR probability estimate and somehow we use it only for the binary case in lda. I feel that the code duplication would be beneficial to understand the code.

This is not for this PR, but I would like to rename `_predict_proba_lr` to `_predict_proba_ovr_lr` such that it is explicit which implementation it is. It has been implemented in the base class just to be able to use it in the `SGDClassifier` which implement an OvR scheme.

In fact, the implementation of the of `predict_proba` could be the same in LDA and logistic (the multinomial formulation). However, the `decision_function` of those 2 classes is a bit different and therefore we cannot reuse it.

@jnothman Could you give your opinion on this matter.

> BTW, I am happy to change `len(self.classes_)` to `self.classes_.size` if this style is preferred.

Since this is a numpy array we can go with `size`.

Reply:
I have no problem with `len`. `size` is also okay...?

Reply:
Oh, you were asking for my opinion on the other matter. I would also accept modifying `_predict_proba_lr` to take `multiclass='multinomial'` or `multiclass='ovr'` as per LogisticRegression

Reply:
So if we want to modify to reuse code we need to modify the `fit` of the discriminant analysis. So basically we want our `predict_proba` to do:

```python
decision = self.decision_function(X)
if decision.ndim == 1:
    decision_2d = np.c_[-decision, decision]
else:
    decision_2d = decision
return softmax(decision_2d, copy=False)
```

Therefore, in the binary case, we need to change `fit` from:

```python
self.coef_ = np.array(self.coef_[1, :] - self.coef_[0, :], ndmin=2)
self.intercept_ = np.array(self.intercept_[1] - self.intercept_[0], ndmin=1)
```

to

```python
self.coef_ = self.coef_[[1]]
self.intercept_ = self.intercept_[1]
```

Otherwise, we can use the softmax in the binary case because we have a multiplicative factor 2 too much.

However, I did not check such change modify something within the transform.

Reply:
Happy to rename `_predict_proba_lr` to `_predict_proba_ovr_lr` if simpler :)